### PR TITLE
test: use soft page refresh in e2e tests

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

closes none

## Description
Switched to using in-app redirects instead of full page reload. This reduced runtime of e2e test suites by ~40%.

Also added 90s timeouts for test cases and 100m timeout for entire test suits

### Before:
image

### After:
image

### CODE REVIEW

File: e2e/wdio/core/pages/multi-input.po.ts [hunk 1]
```diff
@@ -16,12 +16,12 @@ export class MultiInputPo extends CoreBaseComponentPo {
     approveButton = '.fd-button--emphasized';
     multiSelectButton = '.custom-multi-input-select-all-bar-element button';
     mobileInputOptions = 'fd-multi-input-mobile-example .fd-token__text span';
-    displayObjectOptions = 'div#background-ex3 .fd-token__text span';
-    searchTermOptions = 'div#background-ex4 .fd-token__text span';
-    customFilterOptions = 'div#background-ex5 .fd-token__text span';
+    displayObjectOptions = 'fd-multi-input-displaywith-example .fd-token__text span';
```
Comment (InnaAtanasova) [inline e2e/wdio/core/pages/multi-input.po.ts:19]: missed a `-` in `displaywith`

**Review by InnaAtanasova — APPROVED** [review]

File: e2e/wdio/core/pages/multi-input.po.ts (hunk 1 above)
Comment (dmitry-stepanenko) [reply e2e/wdio/core/pages/multi-input.po.ts:19]: @InnaAtanasova it's the actual selector of the component. Do you want me to rename it?
image

Comment (InnaAtanasova) [reply e2e/wdio/core/pages/multi-input.po.ts:19]: no, it's ok. Leave it as it is. Thanks for replying.

File: e2e/wdio/driver/wdio.ts [hunk 2]
```diff
@@ -61,7 +61,21 @@ export function goBack(): void {
...
+export function refreshPage(isFullRefresh = false): void {
+    try {
+        browser.dismissAlert();
+    } catch {}
+    if (!isFullRefresh) {
+        const url = browser.getUrl();
+        click(`#toolbar-home-btn`);
+        if (browser.getUrl().includes(`/home`)) {
+            goBack();
+            return;
```
Comment (platon-rov) [inline e2e/wdio/driver/wdio.ts:73]: If we're in this branch we're omitting `if (browserIsSafari())`, is it intended?

File: e2e/wdio/driver/wdio.ts [hunk 3]
```diff
@@ -61,7 +61,21 @@ export function goBack(): void {
...
+        } else {
+            // failed to navigate, reset the url and perform full page refresh
+            browser.url(url);
```
Comment (platon-rov) [inline e2e/wdio/driver/wdio.ts:76]: If we're in this branch we will be refreshing page 2 times. Here & `browser.refresh();` later.

**Review by platon-rov — COMMENTED** [review]
good job!

File: e2e/wdio/driver/wdio.ts (hunk 3 above)
Comment (dmitry-stepanenko) [reply e2e/wdio/driver/wdio.ts:76]: Nice catch. I was just handling a weird edge cases if any and didn't notice `browser.url` performs normal page redirects. Fixed

File: e2e/wdio/driver/wdio.ts (hunk 2 above)
Comment (dmitry-stepanenko) [reply e2e/wdio/driver/wdio.ts:73]: I'm wondering why they actually added that extra pause for safari, probably it was a bug somewhere, it works fine for me even without it. But it definitely not needed, when we're basically clicking urls on the page, so yes, it's intended.

**Review by platon-rov — APPROVED** [review]

**Pull request merged**